assets/src/components/minimalLadder.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useContext } from "react"
+import RouteLadders from "./routeLadders"
+import { routeTabById } from "../models/routeTab"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import {
+  deselectRouteInTab,
+  flipLadderInTab,
+  toggleLadderCrowdingInTab,
+} from "../state"
+import { useParams } from "react-router-dom"
+
+export const MinimalLadder = () => {
+  const [{ routeTabs }, dispatch] = useContext(StateDispatchContext)
+  const id = useParams().id as string
```

Comment:
Does typescript complain if we do it like this?

```suggestion
  const { id } = useParams()
```

> https://reactrouter.com/en/main/hooks/use-params

Reply:
It would type `id` as a `string | undefined`, which then means that it wouldn't be passable into `routeTabById`.

This is an interesting one, because `id` _should_ always be there because of how our `ReactRouter` is configured, but I wonder how to convince Typescript of that fact.

In theory, 👇 would work, but I'm not sure I love it, given that it's more verbose
```typescript
const { id } = useParams() as { id: string }
```

Reply:
Or re-stated, create a `<MinimalLadder.FromRouterParam>` which wraps `<MinimalLadder>` and does all the `useParams` stuff there (like asserting the id is specified or showing an error message or redirecting otherwise) and then calls (condensed, since you _may_ know via typescript that `id` is defined at this point.)
```
{id && <MinimalLadder id={id} />}
```

Reply:
Yeah that makes sense to me!

Reply:
Ha! I'm not sure why I took the awkward route here. Yes, that's great

Reply:
Omg once again, I didn't refresh to get all the comments. Ignore my last one, I'm re-reading

Reply:
Looking at this with fresh eyes. While creating the `.FromRouterParam` may be the proper way, I don't naturally prefer this approach. I'll do it if I must! But it adds a layer of, like, unneeded padding around the MinimalLadder component that I think won't actually come in handy down the road? It's a layer of extra stuff around MinimalLadder that will add complexity in the App component, to fix one line in MinimalLadder that could be a little tighter. I suppose I find the existing one line more readable. Is that a bad take?

Reply:
I'm not okay with the type coersion (the `as string`), as I feel that's somewhat defeating the usefulnes of typescript, so I think you should handle the `undefined` case explicitly, but however you want to handle that is fine (if you don't feel like making a wrapper component to handle that logic)

Reply:
In terms of reability, I don't find it harder to read, and I find it easier to reason about when the required parameter is just a parameter declared the correct type, but I this this is really just style and taste and nothing to worry about any specific way of doing it.

Reply:
Ohhhkay finally implemented, and I understand it better! Thanks for your patience 🙏